src/main/java/com/cloudbees/hudson/plugins/folder/computed/ChildObserver.java

```diff
@@ -58,4 +63,20 @@
      */
     public abstract void created(I child);
 
+    /**
+     * Returns a copy of the item names that have been observed.
+     *
+     * @return a copy of the item names that have been observed.
+     * @since FIXME
+     */
+    public abstract Set<String> observed();
```

Comment:
Never mind—the constructor was always package access, so adding new `abstract` methods is not incompatible.